main/src/ca/uwaterloo/flix/api/lsp/provider/completion/Completion.scala

```diff
@@ -38,10 +50,47 @@ object Completion {
   /**
     * Represents a keyword completion.
     *
-    * @param name the name of the keyword.
+    * @param name The name of the keyword.
     */
   case class KeywordCompletion(name: String, context: CompletionContext) extends Completion
 
+  /**
+    * Represents a field completion.
+    *
+    * @param name The name of the field.
+    */
+  case class FieldCompletion(name: String, context: CompletionContext) extends Completion
+
+
+  /**
+    * Represents a predicate completion.
+    *
+    * @param name     The name of the predicate
+    * @param priority The priority of the predicate
+    */
+  case class PredicateCompletion(name: String, priority: String, context: CompletionContext) extends Completion
+
+  /**
+    * Represents a BuiltinType completion
+    *
+    * @param name               The name of the BuiltinType
+    * @param priority           The priority of the BuiltinType
+    * @param textEdit           The edit which is applied to a document when selecting this completion.
+    * @param insertTextFormat   The format of the insert text.
+    */
+  case class BuiltinTypeCompletion(name: String, priority: String, textEdit: TextEdit, insertTextFormat: InsertTextFormat) extends Completion
+
+  /**
+    * Represents a Type completion (enums, aliases)
+    *
+    * @param name           The name of the type
+    * @param priority       The priority of the type
+    * @param textEdit       The edit which is applied to a document when selecting this completion.
+    * @param documentation  A human-readable string that represents a doc-comment
+    */
+  case class TypeCompleter(name: String, priority: String, textEdit: TextEdit, documentation: Option[String]) extends Completion
```

Comment:
This name does not make much sense. 

I understand `FieldCompletion`. That's a good name.